java/sage/media/sub/SubtitleHandler.java

```diff
@@ -341,15 +372,22 @@ else if (sage.media.format.MediaFormat.VOBSUB.equals(targetFormat))
   // Returns true if the timing delay for the next subtitle update has changed (i.e. VF needs a kick)
   public boolean postSubtitleInfo(long time, long dur, byte[] rawText, int flags)
   {
-    synchronized (subEntries)
+    if ((flags & FLUSH_SUBTITLE_QUEUE) == FLUSH_SUBTITLE_QUEUE)
     {
-      if ((flags & FLUSH_SUBTITLE_QUEUE) == FLUSH_SUBTITLE_QUEUE)
+      subtitleLock.writeLock().lock();
+
+      try
       {
         subEntries.clear();
         subEntryPos = -1;
       }
-      return insertEntryForPostedInfo(time, dur, rawText);
+      finally
+      {
+        subtitleLock.writeLock().unlock();
+      }
     }
+
+    return insertEntryForPostedInfo(time, dur, rawText);
```

Comment:
I was trying to remove the need to lock at that spot if it wasn't needed. I took at a look at how the parameters are actually used and determined that we didn't need to get the lock, but to make things closer to as they were I changed it per your suggestion.